cmd/upgrade.go

```diff
@@ -89,7 +87,7 @@ func (d *diffCmd) isAllowUnreleased() bool {
 //
 // See also https://github.com/helm/helm/pull/9426#discussion_r1181397259
 func (d *diffCmd) clusterAccessAllowed() bool {
-	clientOnly := (d.dryRunModeSpecified && d.dryRunMode == "") || d.dryRunMode == "true" || d.dryRunMode == "client"
+	clientOnly := d.dryRunMode == "true" || d.dryRunMode == "client"
```

Comment:
We could do better, but the intention is; we ensure that `dryRunMode` ever takes any of `true`, `false`, `client` and `server` by doing `if diff.dryRunMode != "true" && diff.dryRunMode != "false" && diff.dryRunMode != "client" && diff.dryRunMode != "server" {` check in `cmd/upgrade.go`.

It's enough to check for `true` and `client` to say it's `clientOnly`.
If the user didn't told us to be `clientOnly`, it implies `clusterAccessAllowed`.